python_modules/dagster/dagster/core/execution/execute_in_process_result.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Dict, List, Optional
+
+from dagster import DagsterEvent, check
+from dagster.core.definitions import NodeDefinition, NodeHandle
+from dagster.core.definitions.utils import DEFAULT_OUTPUT
+from dagster.core.errors import DagsterInvariantViolationError
+from dagster.core.execution.plan.outputs import StepOutputHandle
+
+
+class InProcessResult:
+    def __init__(
+        self,
+        node_def: NodeDefinition,
+        all_events: List[DagsterEvent],
+        run_id: str,
+        output_capture: Optional[Dict[StepOutputHandle, Any]],
+    ):
+        self._node_def = node_def
+
+        # If top-level result, no handle will be provided
+        self._handle = NodeHandle(node_def.name, parent=None)
+        self._event_list = all_events
+        self._run_id = run_id
+
+        self._output_capture = output_capture
+
+    @property
+    def success(self):
+        """bool: Whether execution was successful."""
+        return all([not step_event.is_step_failure for step_event in self.all_node_events])
```

Comment:
some rare cases of pipeline failure happen without step failures

Reply:
Yea this made me uneasy. Gonna have to do something smarter here.